### PR TITLE
Reviewing questionnaire - Approving

### PR DESCRIPTION
**Pull request by heofs**
### What is the context of this PR?
This PR introduces reviewing. When a user submits a questionnaire for approval, an admin should then be able to approve it. Approval will then move it into the survey register. 

### How to review 
1. Be logged in as an admin
1. Create a questionnaire and submit it for approval in the publish page
1. Publish button should now say review.
1. Log into a user that is not an admin, and see that there is a red bar at the top saying the questionnaire is being reviewed and can not be edited.
1. Check the homepage and see that the status has changed to "Awaiting Approval".
1. Make sure you can not edit the questionnaire.
1. To review, log in as admin, click the review button, and click `Approve`.
1. Red box at the top should now disappear, and questionnaire list should now say the questionnaire status is `Published`.

### CODE REVIEW

File: eq-author/src/App/QuestionnairesPage/QuestionnairesView/QuestionnairesTable/TableHead/index.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ const TableHead = props => {
       <tr>
         <SortableTH
           sortColumn="title"
-          colWidth="18%"
+          colWidth="27%"
```
Comment (SamGodwin2) [inline eq-author/src/App/QuestionnairesPage/QuestionnairesView/QuestionnairesTable/TableHead/index.js]: Can we remove a few percent from here and add it to permissions, that columns looking a little tight.

File: eq-author/src/components/buttons/Button/index.js [hunk 2]
```diff
@@ -97,6 +111,7 @@ const Button = styled.button`
   ${props => props.variant === "secondary" && secondaryButton};
   ${props => props.variant === "tertiary" && tertiaryButton};
   ${props => props.variant === "tertiary-light" && tertiaryLightButton};
+  ${props => props.variant === "success" && successButton};
```
Comment (SamGodwin2) [inline eq-author/src/components/buttons/Button/index.js]: As much as I like the idea of a "success"! button 😂 is it worth adding a new one here or can we just overwrite the colors of one of the other buttons?

File: eq-author/src/App/QuestionnairesPage/QuestionnairesView/QuestionnairesTable/TableHead/index.js (hunk 1 above)
Comment (heofs) [reply eq-author/src/App/QuestionnairesPage/QuestionnairesView/QuestionnairesTable/TableHead/index.js]: Added a bit to permission instead

**Comment by heofs** [comment]
[Approve content](https://trello.com/c/ExJa5ANr/1199-approve-content)

File: eq-author/src/components/EditorLayout/Header/index.js [hunk 3]
```diff
@@ -95,12 +96,45 @@ export class UnconnectedHeader extends React.Component {
     this.setState({ isQuestionnaireSettingsModalOpen: true });
   };
 
+  renderPublishReviewButton = () => {
+    const { questionnaire, title, match } = this.props;
+    const publishStatus = questionnaire && questionnaire.publishStatus;
+    if (publishStatus === "AwaitingApproval") {
```
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.js]: This is on the frontend, it doesnt have a constant for it. I can create a new constant for it as it is used multiple places.

File: eq-author/src/components/EditorLayout/Header/index.js [hunk 4]
```diff
@@ -95,12 +96,45 @@ export class UnconnectedHeader extends React.Component {
...
+    }
+    return (
+      <RouteButton
+        variant="tertiary-light"
+        to={buildPublishPath(match.params)}
+        small
+        disabled={
+          questionnaire.totalErrorCount > 0 ||
+          title === "Publish" ||
+          publishStatus === "Published"
```
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.js]: This too is on the frontend

File: eq-author/src/components/EditorLayout/Header/index.js [hunk 5]
```diff
@@ -95,12 +96,45 @@ export class UnconnectedHeader extends React.Component {
...
+      >
+        <IconText icon={publishIcon}>Publish</IconText>
+      </RouteButton>
+    );
+  };
+
   render() {
     const { questionnaire, title, children, me, client } = this.props;
     const previewUrl = `${config.REACT_APP_LAUNCH_URL}/${
       (questionnaire || {}).id
```
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.js]: This is not related to this PR but I don't think that line can error.
If questionnaire is passed into the props as `undefined`. And then you do `(undefined || {}).id`, that will return `undefined`. But if `questionnaire === undefined` and you do `undefined.id`, That will throw an `Uncaught TypeError: Cannot read property 'id' of undefined`.

File: eq-author/src/components/EditorLayout/Header/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.js]: Adding the constant now :)

File: eq-author/src/components/BaseLayout/PermissionsBanner.js [hunk 6]
```diff
@@ -42,26 +45,39 @@ const WarningMessage = styled(IconText)`
   padding: 1em;
 `;
 
+const warningMessages = {
+  editorAccess:
+    "You do not have editor access to this questionnaire, any changes you make will not be saved",
+  awaitingApproval:
+    "This questionnaire is currently being reviewed. Any changes made will not be saved.",
```
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author/src/components/BaseLayout/PermissionsBanner.js]: I agree, but I have spoken with Sarah and they can be kept that way.

File: eq-author-api/schema/tests/questionnaire.test.js [hunk 7]
```diff
@@ -136,6 +149,105 @@ describe("questionnaire", () => {
...
+          reviewQuestionnaire(
+            {
+              questionnaireId: ctx.questionnaire.id,
+              reviewAction: "Approved",
+            },
+            ctx
+          )
+        ).rejects.toBeTruthy();
+        expect(ctx.questionnaire.publishStatus).toEqual(AWAITING_APPROVAL);
+      });
```
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author-api/schema/tests/questionnaire.test.js]: Here I am testing that the promise is rejected. But getting the error message is a pain :/

**Pull request merged**